### PR TITLE
Issue 19845

### PR DESCRIPTION
**Pull request by erickgonzalez**
Check if the current host request attribute is set before doing a search of templates, if is set use it as a filter.

### CODE REVIEW

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/template/TemplateResourceTest.java [hunk 1]
```diff
@@ -992,6 +993,35 @@ public void test_listTemplate_filterByHost()
...
+        //Create Request and set Attribute
+        final HttpServletRequest request = getHttpRequest(adminUser.getEmailAddress(),"admin");
+        request.getSession().setAttribute(WebKeys.CURRENT_HOST,newHostA);
+        //Call Resource
+        final Response responseResource = resource.list(request,response,"",0,40,"mod_date","DESC","",false);
+        //Check that the response is 200, OK
+        Assert.assertEquals(Status.OK.getStatusCode(),responseResource.getStatus());
+        final ResponseEntityView responseEntityView = ResponseEntityView.class.cast(responseResource.getEntity());
+        final PaginatedArrayList paginatedArrayList = PaginatedArrayList.class.cast(responseEntityView.getEntity());
+        Assert.assertEquals(1,paginatedArrayList.size());
```
Comment (nollymar) [inline dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/template/TemplateResourceTest.java:1022]: Can you validate if the returned template is actually the one related to the current host?

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/template/TemplateResource.java [hunk 2]
```diff
@@ -159,8 +160,9 @@ public final Response list(@Context final HttpServletRequest httpRequest,
         final InitDataObject initData = new WebResource.InitBuilder(webResource)
                 .requestAndResponse(httpRequest, httpResponse).rejectWhenNoUser(true).init();
         final User user = initData.getUser();
+        final Lazy<String> lazyCurrentHost = Lazy.of(() -> Try.of(() -> Host.class.cast(httpRequest.getSession().getAttribute(WebKeys.CURRENT_HOST)).getIdentifier()).getOrNull());
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/template/TemplateResource.java:163]: Is it worth to add a test for the case when `lazyCurrentHost` is null?

Comment (erickgonzalez) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/template/TemplateResource.java:163]: There is already a postman test where check this case, no current host set neither the query param.

File: dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/template/TemplateResourceTest.java (hunk 1 above)
Comment (erickgonzalez) [reply dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/template/TemplateResourceTest.java:1022]: done

**Review by nollymar — APPROVED** [review]

**Pull request merged**